Some minor overhauls

richteer opened the pull request:
See the commit messages for each.

Important notes:
- tests will probably have to be update
- I wrote these on a plane, there's a good chance they are buggy

richteer committed:
loader: implement new format for loading module classes

Previously, the loader scanned loaded module files for classes
that inherited the halibot.HalModule class. This was not only
inefficent, but also buggy with further subclasses (e.g.
halibot.CommandModule) and rather unpythonic.

The new specifier in a module's config.json is as follows:
	"main": "file.py:ModuleName"

The loader will attempt to split on ':', and if successful,
load the first as the module file, and the latter of a HalModule.
There is no longer any silly type checking, so not inheriting from
HalModule is totally possibly, but probably not a good idea.

The old loading style still remains for if the 'main' key can't be
split, but should probably be removed in the future.

richteer committed:
main: remove useless 'imp' import

richteer committed:
main: add command line arguments, add some instance init

Slight overhaul of the main halibot entry point. Includes the following
arguments:
	init - Create a template config.json, modules, and agents dir
	run  - Run a halibot instance using the local dir's config
	help - Show the help.

tophj commented on main.py:
```diff
@@ -1,13 +1,88 @@
+	else: # Dir does exist, ensure clear
+		if os.listdir(path):
+			r = input("Directory not empty, are you sure you want to init here? [y/N]: ")
+			if not r.lower() in ("y", "yes", "yolo"):
```
+1 to yolo init

sjrct commented:
Update the configs in the `*container` and the `irc` agent.

sjrct commented on halibot/loader.py:
```diff
@@ -28,31 +28,58 @@ def _find(self, name):
+		try:
+			obj = getattr(pymod,clsname)
+		except:
+			self.warning("Module '{}' failed to load: Could not find class '{}' in '{}'".format(name, src, cls))
```
I think this should be an error.

sjrct commented on halibot/loader.py:
```diff
@@ -28,31 +28,58 @@ def _find(self, name):
-		src = basepath + config["main"]
+		# Check if using the new loader format or not
+		if not config.get("main"):
+			self.log.warning("Failed to load '{}', 'main' key not found in module config".format(name))
```
Should be an error.

sjrct commented:
Looks pretty mergable after that.

sjrct committed:
Update to use new module/agent config format

Update configs of:
 * irc
 * agentcontainer
 * modulecontainer

sjrct committed:
Change levels of some loader logs

Things that cause a module to not be loaded should be an error, as
Halibot will not behave as expected (i.e. use the specified module) in
those cases. Also, non-breaking behavoir that is deprecated and can
therefore be removed in the future should be a warning.
